datafusion/sqllogictest/test_files/group_by.slt

```diff
@@ -3425,7 +3425,7 @@ physical_plan
 06)----------RepartitionExec: partitioning=Hash([sn@0, amount@1], 8), input_partitions=8
 07)------------AggregateExec: mode=Partial, gby=[sn@0 as sn, amount@1 as amount], aggr=[]
 08)--------------RepartitionExec: partitioning=RoundRobinBatch(8), input_partitions=1
-09)----------------DataSourceExec: partitions=1, partition_sizes=[1]
+09)----------------DataSourceExec: partitions=1, partition_sizes=[2]
```

Comment:
I believe the reason the partition sizes increases is that `sales_global_with_pk` is created from the output if a query , and since the batch size is 4, this results in more batches from the input.

I think this is a better plan overall